### PR TITLE
Issue1437 1662 ets direct cooling

### PR DESCRIPTION
**Pull request by khinkelman**
Hi @JayHuLBL, here are two new direct cooling ETS models with their example files. These models are based on typical designs described in ASHRAE's District Cooling Guide. Please let me know if you have any questions/comments. Thanks!

### CODE REVIEW

**Comment by amyeallen1** [comment]
Thanks, Katy! This is a small tweak, but could the mass flow rate and supply and return temperatures could be exposed as outputs, in both these ETS models? It would be good if the component models could generally have the output variables in this list, as applicable: https://github.com/urbanopt/geojson-modelica-translator/blob/develop/README.rst
@khinkelman

**Comment by khinkelman** [comment]
Hi @amyeallen1, per our conversation today: These variables are public within the model and therefore are available for reporting, even though they are not brought to the model's top layer as outputs. Let me know if you want to meet on campus to look at the model and simulation results for extra clarity.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+<img src=\"modelica://Buildings/Resources/Images/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.PNG\" alt=\"DHC.ETS.CoolingDirectControlledReturn\"/>
+</p>
+<h4>Reference</h4>
+<p>
+American Society of Heating, Refrigeration and Air-Conditioning Engineers. 
+(2013). Chapter 5: End User Interface. In <i>District Cooling Guide</i>. 1st Edition. 
+</p>
+</html>", revisions="<html>
+<ul>
+<li>December 12, 2019, by Kathryn Hinkelman:<br>First implementation. </li>
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: The `<br>` should be `<br/>`.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+<img src=\"modelica://Buildings/Resources/Images/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.PNG\" alt=\"DHC.ETS.CoolingDirectUncontrolled\"/>
+</p>
+<h4>Reference</h4>
+<p>
+American Society of Heating, Refrigeration and Air-Conditioning Engineers. 
+(2013). Chapter 5: End User Interface. In <i>District Cooling Guide</i>. 1st Edition. 
+</p>
+</html>", revisions="<html>
+<ul>
+<li>November 13, 2019, by Kathryn Hinkelman:<br>First implementation. </li>
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo]: The `<br>` should be `<br/>`.

File: Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectControlledReturn.mo [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+<p>
+This model provides an example for the direct cooling energy transfer station 
+model, which contains in-building pumping and controls the district return 
+temperature. The building's primary variable speed pump is modulated depending 
+on the total cooling load and prescribed deltaT. Variation in the district 
+supply temperature is modeled as sinusoidal to test the system's response. 
+</p>
+</html>", revisions="<html>
+<ul>
+<li>December 12, 2019, by Kathryn Hinkelman:<br>First implementation. </li>
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectControlledReturn.mo]: The `<br>` should be `<br/>`.

File: Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectUncontrolled.mo [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+<p>
+This model provides an example for the direct cooling energy transfer station 
+model, which does not contain in-building pumping or deltaT control. The 
+ultimate control lies with the thermostatic control valve at the lumped, 
+terminal building load. The control valve is modulated proportionally to the 
+instantaneous cooling load with respect to the maxiumum load.
+</p>
+</html>", revisions="<html>
+<ul>
+<li>December 5, 2019, by Kathryn Hinkelman:<br>First implementation. </li>
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectUncontrolled.mo]: The `<br>` should be `<br/>`.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+    "Product"
+    annotation (Placement(transformation(extent={{-10,100},{10,120}})));
+
+  Modelica.Blocks.Math.Gain cp(final k=cp_default)
+    "Specific heat multiplier to calculate heat flow rate"
+    annotation (Placement(transformation(extent={{30,100},{50,120}})));
+
+  Buildings.Fluid.FixedResistances.Junction spl(
+    redeclare final package Medium = Medium,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: Propagate the parameter `energyDynamics` and put it in the `Advanced` tab. The same `energyDyanmics` is also used by instance `jun`.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+    final m_flow_nominal={mBui_flow_nominal,-mDis_flow_nominal,-mByp_flow_nominal},
+    dp_nominal=500*{1,-1,1}) "Bypass junction, splitter"
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: Also propagate `dp_nominal` and put it into `Advanced` tab.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+        rotation=90,
+        origin={40,0})));
+
+  Modelica.Blocks.Sources.Constant ope(final k=1)
+    "Check valve is always open in the positive flow direction"
+    annotation (Placement(transformation(extent={{0,-10},{20,10}})));
+
+  Buildings.Controls.Continuous.PIDHysteresis con(
+    eOn=0,
+    eOff=-0.5,
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: should `eOn` and `eOff` be tunning parameters that user can change? If they are, then they should be propagated.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+    annotation (Line(points={{70,-49},{70,-40},{54,-40},{54,-140},{-50,-140},{-50,-132}},
+      color={0,0,127}));
+  connect(one.y, add.u1)
+    annotation (Line(points={{-59,-94},{-22,-94}}, color={0,0,127}));
+  connect(con.y, add.u2)
+    annotation (Line(points={{-39,-120},{-30,-120},{-30,-106},{-22,-106}},
+      color={0,0,127}));
+  connect(add.y, conVal.y)
+    annotation (Line(points={{1,-100},{10,-100},{10,-72}}, color={0,0,127}));
+
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: remove this empty line

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  parameter Modelica.SIunits.MassFlowRate m_flow_nominal(
+    final min=0,
+    final start=0.5)
+    "Nominal mass flow rate";
+
+  // pressure drops
+  parameter Modelica.SIunits.PressureDifference dpSup(
+    final min=0,
+    displayUnit="Pa")=50
+  "Pressure drop in the ETS supply side";
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo]: use a higher default value. `50 pa` seems to be a very short section of pipe.

File: Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectControlledReturn.mo [hunk 10]
```diff
@@ -0,0 +1,167 @@
+within Buildings.Applications.DHC.EnergyTransferStations.Examples;
+model CoolingDirectControlledReturn
+  "Example model for direct cooling energy transfer station with in-building 
+  pumping and controlled district return temperature"
+  extends Modelica.Icons.Example;
+
+  package Medium = Buildings.Media.Water;
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectControlledReturn.mo]: add comment

File: Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectUncontrolled.mo [hunk 11]
```diff
@@ -0,0 +1,157 @@
+within Buildings.Applications.DHC.EnergyTransferStations.Examples;
+model CoolingDirectUncontrolled
+  "Example model for the direct cooling energy transfer station with 
+  uncontrolled district-building fluid transfer within the ETS"
+  extends Modelica.Icons.Example;
+
+  package Medium = Buildings.Media.Water;
```
Comment (JayHuLBL) [inline Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingDirectUncontrolled.mo]: add comment

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo (hunk 7 above)
Comment (khinkelman) [reply Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo]: Yes, I don't see why not. I'll propagate these as well.

File: Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo (hunk 9 above)
Comment (khinkelman) [reply Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectUncontrolled.mo]: It is. The pipe length (in the mechanical room) for a direct connection ETS like this shouldn't be too long, although I agree this default may be too low. I'll adjust after some additional research.

**Pull request merged**